### PR TITLE
Fix various spelling mistakes

### PR DESCRIPTION
**Pull request by neonjonn**
### Summary

Fix various spelling mistakes across 15 files.
This is a work in progress I haven't checked *everything* yet!

And for those wondering why I'm still coding instead of focusing on school, I'm on holiday! :<

### Checklist

- [ ] I have run `nox` and all the pipelines have passed.
    * verify-types failed, something to do with the environment already existing 🤷‍♂️
- [ ] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: docs/config.mako [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
     hljs_style = "atom-one-dark"
 
-    # Doesn't really do anything, just enables lurn_search
+    # Doesn't really do anything, just enables lunr_search
```
Comment (davfsa) [inline docs/config.mako:38]: You can leave this one out, it's being removed anyways :)

File: .github/workflows/prepare-release.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ on:
         required: true
 
 jobs:
-  prepare-relese:
+  prepare-release:
```
Comment (neonjonn) [inline .github/workflows/prepare-release.yml:11]: I've no idea if changing the job name will break anything, so please let me know if this needs to be reverted.

**Review by davfsa — COMMENTED** [review]
Tysm. Going to be fun to rebase my docs branch against this :))

If you don't mind me asking, where you using some tool for these or just manual search?

File: .github/workflows/prepare-release.yml (hunk 2 above)
Comment (davfsa) [reply .github/workflows/prepare-release.yml:11]: Doesn't break anything, no worries :)

**Comment by neonjonn** [comment]
> If you don't mind me asking, where you using some tool for these or just manual search?

I was using [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) for VS Code, but I go through each change it suggests manually... I haven't dared look at the tests yet after opening one of them and getting 50+ warnings about purposely made mistakes :wheeze:

File: docs/config.mako (hunk 1 above)
Comment (neonjonn) [reply docs/config.mako:38]: oh okay, will do!

**Comment by davfsa** [comment]
> Looking through those tests was certainly an experience, I must say. I especially liked `test_snowflake` 🤣 (*cough* neko *cough*)
> 
> Edit: I've gone through all the files now, I think that's all the spalling mistakes fixed :)

Tysm for all of it. All looks nice :)

That spelling mistake in the CHANGELOG I will also have to fix in the GitHub release, so leaving myself a todo for that.

Thanks again!

**Comment by neonjonn** [comment]
No problem, even if I can't code well enough to make a good contribution, I'll fix grammar where I can!

**Review by davfsa — APPROVED** [review]

**Pull request merged**